freeipa/src/main/java/com/sequenceiq/freeipa/service/config/RootCertRegisterService.java

```diff
@@ -68,6 +63,31 @@ public RootCert register(Stack stack) throws FreeIpaClientException {
         return registerInternal(stack);
     }
 
+    @Override
+    public void delete(Stack stack) {
+        try {
+            rootCertService.deleteByStack(stack);
+        } catch (NotFoundException e) {
+            LOGGER.info("Root cert not exists for environment {}", stack.getEnvironmentCrn());
+        }
+    }
+
+    private RootCert registerInternal(Stack stack) throws FreeIpaClientException {
+        MDCBuilder.buildMdcContext(stack);
+        String rootCertificate = getRootCertFromFreeIpa(stack);
+        Optional<RootCert> currentCert = rootCertService.findByStackId(stack.getId());
+        return currentCert.isPresent() ? updateRootCertEntity(currentCert.get(), rootCertificate) : createRootCertEntity(stack, rootCertificate);
+    }
+
+    private RootCert updateRootCertEntity(RootCert rootCert, String rootCertificate) {
```

Comment:
nit: naming could be improved a bit
`updateRootCertEntityIfDifferent`or similar